src/net/ossl.cc

```diff
@@ -315,11 +315,9 @@ class tls::certificate_credentials::impl {
         if (!pkey) {
             throw ossl_error("Error attempting to parse private key");
         }
-#if 0 // https://github.com/redpanda-data/core-internal/issues/1233
-        if (!X509_verify(x509_cert.get(), pkey.get())) {
+        if (!X509_check_private_key(x509_cert.get(), pkey.get())) {
```

Comment:
Can a test be added?

Reply:
Seastar's TLS unit tests do verify the coherency of the key and cert